### ISSUE TITLE
Convert EDTF/Gregorian calendar dates to Julian calendar dates?

### ISSUE THREAD

**Issue by nickbart1980** [plk/biblatex#448]
Here’s an interesting one:

Would it be possible to optionally convert [Gregorian calendar dates](https://en.wikipedia.org/wiki/Gregorian_calendar) to [Julian calendar dates](https://en.wikipedia.org/wiki/Julian_calendar)?

EDTF is an excellent standardised *input* format based on the Gregorian calendar with a year 0, and can be used as an unequivocal representation of *any* date, including dates before the first introduction of the Gregorian calendar on 15 October 1582 (so-called [proleptic Gregorian calendar](https://en.wikipedia.org/wiki/Proleptic_Gregorian_calendar)).

However, users might wish to *output* Julian calendar dates instead. In fact, most historical publications, both popular and academic, use Julian calendar dates for events before 15 Oct 1582 (Gregorian).

For example, Augustus’ birth and death are correctly represented by the EDTF string “-0062-09-21/0014-08-17”, but more usually given as “23 Sep 63 BCE–19 Aug 14 CE (Julian)”.

Or for something bibliographically more relevant, most users would want a book with an EDTF date of “1493-07-21” to be output with the customary Julian calendar date:

Schedel, Hartmann (12 July 1493). *Liber Chronicarum.* Nuremberg: Anton Koberger.

For events in territories that did not adopt the Gregorian calendar at once, e.g., Britain and its Empire between 15 Oct 1582 (Gregorian) and 14 Sep 1752 (Gregorian), the Julian, or the Gregorian, or both calendar dates may be given (see [Dual dating](https://en.wikipedia.org/wiki/Dual_dating) and [Old Style and New Style dates](https://en.wikipedia.org/wiki/Old_Style_and_New_Style_dates)).

So, what biblatex could do is:

- Optionally convert input dates with day precision that lie before a certain cut-off date (default could be 15 Oct 1582) to its corresponding date in the Julian calendar.

- Optionally add a label such as “(Julian)”.

- Optionally output *both* dates, with appropriate labels.

- However, I would not bother with trying to convert dates with year, month, or season precision: Most events we will in all likelihood be able to date with day precision will have occurred, say, later than –1500, and for these, the difference between the Julian and the Gregorian calendar will be at most 13 days, i.e., less than half a month. – EDIT: If we accept the eclipse of Thales, EDTF −0584-05-22 = 28 May 585 BCE Julian as the earliest historical event whose date is known with day precision, the maximum practically relevant difference will actually be the 13 days’ difference between the calendars *currently* in effect (to be precise, from 01 Mar 1900 to 28 Feb 2100 (Greg.)).

Finally, why not resort to a hack and include the intended Julian dates themselves (rather than their Gregorian counterparts) in EDTF strings (e.g., “-0062-09-23/0014-08-19” for Augustus)? – Quite simply: That’s not EDTF-conformant, and there are some Julian dates, e.g., the leap day 29 Feb 1300 (Julian) for which no such hack works (at least not reliably, and not if a strict EDTF parser is involved) since “1300-02-29” does not exist in the Gregorian calendar and hence is not a valid EDTF date.

**Comment by plk** [comment]
Please try latest 3.5/2.6 uploads. See the PDF manual for the new `julian`, `julianstart` and `julianend` options. I added a "start" option in case people don't want to apply Julian date conversion backwards in time forever (before it even existed, for example ... perhaps astronomical dates are preferable in such cases). Such conversion will be ignored for "dates" which are missing months and days since this makes no sense due to ambiguity (example is in the manual). Providing both Julian and Gregorian/astronomical for a date is a bit of a pain as it would need a whole slew of new internal fields.

There is no provision at the moment for output formatting although there are tests for every date field automatically created for testing if a date is Julian or not (`\ifdatejulian`). I'm not sure things like `(Julian)` belong in standard styles anyway?

**Comment by nickbart1980** [comment]
Fantastic. (I’m not actually seeing any Julian dates in my output yet – so, just to confirm, downloading the new biber binary from, e.g.,  https://sourceforge.net/projects/biblatex-biber/files/biblatex-biber/development/binaries/OSX_Intel/biber-darwin_x86_64.tar.gz/download, unpacking it, and placing it somewhere in my path is all I need to do on the biber side?)

> I'm not sure things like (Julian) belong in standard styles anyway?

The convention among most historians seems to be to tacitly assume Julian for any date before 1582-10-15 (Gregorian), but between 1582 and 1752 in Europe and North America (and 1582–1918 in Russia), I’ve seen various forms of labels being used (see, e.g., the dates of Newton’s birth and death in https://fr.wikipedia.org/wiki/Isaac_Newton or https://de.wikipedia.org/wiki/Isaac_Newton – not in https://en.wikipedia.org/wiki/Isaac_Newton, however!). Still, I’d agree it’s not a top priority, and neither is double dating.

**Comment by plk** [comment]
On the biber side, that is all you need to do but you also need to update biblatex with the latest 3.5 download to get the new options. `julian=true` is all you need to get the default of Julian for things before 15 Oct 1582. Let me know of any problems - it's not tested a lot yet.

**Comment by nickbart1980** [comment]
> … you also need to update biblatex with the latest 3.5 download …

That’s clear.

Still, this is what I currently get:

~~~
\documentclass{article}
\usepackage[style=authoryear, dateera=secular, alldates=long, julian=true]{biblatex}
\usepackage{fontspec}
\setmainfont{Linux Libertine}
\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@online{      newton,
  author    = {Newton, Isaac},
  date      = {1643-01-04/1727-03-31},
  title     = {Life: 4 Jan 1643 G – 31 Mar 1727 G = 25 Dec 1642 J – 20 Mar 1727 J},
}
\end{filecontents}
\addbibresource{\jobname.bib}
\begin{document}

\section*{Text}

Blah \autocite{newton}.

\printbibliography

\end{document}
~~~

![screen shot 2016-06-28 at 11 17 48](https://cloud.githubusercontent.com/assets/2765811/16412025/a8404e62-3d19-11e6-8dbe-09637aad3205.png)

**Comment by plk** [comment]
Right - did I misunderstand? The default for `julianend` is 1582-10-15 and since neither of those dates are before this, they are not converted.

**Comment by nickbart1980** [comment]
See my EDIT above. I had tried it with `1400-01-01` before, which was not converted to Julian either. Here’s that MWE:

~~~
\documentclass{article}
\usepackage[style=authoryear, dateera=secular, alldates=long, julian=true]{biblatex}
\usepackage{fontspec}
\setmainfont{Linux Libertine}
\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@online{      doe,
  author    = {Doe, John},
  date      = {1400-01-01},
  title     = {Title},
}
\end{filecontents}
\addbibresource{\jobname.bib}
\begin{document}

\section*{Text}

Blah \autocite{doe}.

\printbibliography

\end{document}
~~~

![screen shot 2016-06-28 at 12 08 56](https://cloud.githubusercontent.com/assets/2765811/16413314/c4d3b206-3d20-11e6-9347-f268590b4181.png)

Can you reproduce this?

**Comment by plk** [comment]
Sorry, stray bracket in the wrong place in biber which made it not work without `julianstart` defined. Should be fixed if you download biber 2.6 again.

**Comment by nickbart1980** [comment]
Great. Seems to work nicely now.

**Comment by nickbart1980** [comment]
Another issue:

EDTF output must never be converted to Julian (see example), even if `julian=true`.

In other words, `alldates=edtf` etc. must enforce `julian=false`.

EDTF *by definition* exclusively uses the Gregorian calendar (proleptic for dates < 1582-10-15) – and allowing output dates that superficially look like EDTF but in fact aren’t is just a recipe for utter confusion.

~~~
\documentclass{article}
\usepackage[style=authoryear, alldates=edtf, julian=true, julianend=1752-09-14]{biblatex}
\usepackage{fontspec}
\setmainfont{Linux Libertine}
\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@online{      newton,
  author    = {Newton, Isaac},
  date      = {1643-01-04/1727-03-31},
  title     = {Life: 4 Jan 1643 G – 31 Mar 1727 G = 25 Dec 1642 J – 20 Mar 1727 J},
}
\end{filecontents}
\addbibresource{\jobname.bib}
\begin{document}

\section*{Text}

Blah \autocite{newton}.

\printbibliography

\end{document}
~~~

![screen shot 2016-06-29 at 14 39 05](https://cloud.githubusercontent.com/assets/2765811/16454153/12462530-3dff-11e6-96fe-f604b6680f21.png)

**Comment by nickbart1980** [comment]
Also, the current `julianend` default is wrong: It must be `1582-10-14`, not `1582-10-15` – or else `1582-10-15` (the first day of the Gregorian calendar) itself is converted to Julian.

However, I feel `gregorianstart` (with `1582-10-15` as a correct default) would actually be a better choice than `julianend`.

After all, it’s typically the first day of the new Gregorian calendar that is given in textbooks and other docs rather than what `julianend` amounts to, “the day before this, in the proleptic Gregorian calendar”.

In addition, I do not think `julianstart` makes too much sense. I’m pretty sure there is not a single calendar that has ever switched *from* Gregorian *to* Julian. For example, astronomers *do* use Julian calendar dates for *any* date before 1582-10-15 (see, e.g., http://eclipse.gsfc.nasa.gov/SEcat5/SEcatalog.html).

**Comment by plk** [comment]
That's fair enough - should all be fixed/changed (needing both biblatex and biber updates).

**Comment by nickbart1980** [comment]
Great – seems to work very well now.

**Comment by JohnLukeBentley** [comment]
Thanks for working through this thorny issue Nick.

> most users would want a book with an EDTF date of “1493-07-21” to be output with the customary Julian calendar date:

> Schedel, Hartmann (12 July 1493). Liber Chronicarum. Nuremberg: Anton Koberger.

@nickbart1980 wrote in [Circa dates, circa date ranges, and question marked dates. Plus Eras. #427 ](https://github.com/plk/biblatex/issues/427#issuecomment-231528050)
> \6. Julian options. No country has ever switched from Gregorian to Julian, and virtually no country uses Julian any longer (very few exceptions: Berbers, Mount Athos). So I don’t expect much demand for an all-Julian calendar, but if it’s really needed, gregorianstart=9999-12-31 or so should work.

> [Nick quoting me (JohnLukeBentley):]
>>  the assumption that in most bibliographic contexts the gregorian calendar is used for all dates, including those before 1582-10-15 (is this true?)

> Not quite. Most historians, and astronomers use Julian for events before 1582-10-15 G. (And most British historians use Julian [or dual dating] for events in Britain etc. before 1752-09-14 G.)

By (most) "historians" do you mean those specifically creating a paper on history or do you mean to include those, say, in the department of philosophy who, when providing bibliographic dates before 1582-10-15, are also acting as "historians"?

More simply: which calendar will philosophers conventionally use for bibliographic dates before 1582-10-15 (the answer defies my google-fu)?

**Comment by nickbart1980** [comment]
> More simply: which calendar will philosophers conventionally use for bibliographic dates before 1582-10-15 (the answer defies my google-fu)?

To the best of my knowledge, Julian.

**Comment by JohnLukeBentley** [comment]
Thanks Nick. 

I've gone fishing for any further information in /r/askphilosophy, posted as [For works and events referenced in philosophy articles which calendar is conventionally used for dates prior to 1582-10-15, the Julian or Gregorian?](https://www.reddit.com/r/askphilosophy/comments/4snnw3/for_works_and_events_referenced_in_philosophy/) (and I've drawn on much of your analysis in your initial post).

I imagine you are right that for dates that are precise only to the level of the month (or less precise): Julian to Gregorian conversions aren't going to matter. I suspect the number of works written prior to 1582-10-15 that are known to be published on a particular day are very few. Therefore I suspect that Julian to Gregorian conversions for the date of publication of works, for works before 1582-10-15: is going to be rare.

**Comment by JohnLukeBentley** [comment]
## Gregorian to Julian conversions

No response from reddit.

I'm inclined to suppose, for output formats:

* As I mentioned, that Julian to Gregorian conversions for the date of publication of works, for works before 1582-10-15: is going to be rare. As precisions at the day level will be rare.
* In the absence of countervailing information, in line with the best of Nick's knowledge, that conversions to Julian, when and where a Julian calendar was in effect, applies for all subject areas (e.g. including philosophy).

## Two date issues: Julian/Gregorian and Start-of-year

The issues do get thornier ...

Nick linked to Old Style V New Style dates and Dual Dating. The best treatment of this might be [Mike Spathaky, 2011-12-13 (Last Updated), Old Style and New Style Dates and the change to the Gregorian Calendar: A summary for genealogists](http://www.cree.name/genuki/dates.htm), (as well as https://en.wikipedia.org/wiki/Old_Style_and_New_Style_dates and  https://en.wikipedia.org/wiki/Dual_dating ).

What emerges is that at various times in the late second millennium there were two issues with dates:

*  Whether the date is in Julian or Gregorian; and (as a separate matter) ...
*  When the beginning of the year is figured: either from 1st Jan (aka "The Historical Year") or some other date (aka "The Civil, or Legal, Year", e.g. For some while the British used 25th March).

## Start-of-year conversions

So take the British case, at a time when both the Julian calendar was in use and the start-of-year was figured to be begin on the 25th March. 

Using my own conventions for representation for the moment, the following reference the same date (SOYMMDD = start-of-year by month and day) ...

* ` 3rd March 1734 (SOY0101 Julian)` 
* ` 3rd March 1733 (SOY0325 Julian)` (Without the disambiguiators this would probably be the representation in the source document. The Civil start of year has not yet kicked on, so we are still in '1733' in Civil year terms, but we are in '1743' in Historical year terms.)
* `14th March 1734 (SOY0101 Gregorian)`
* `14th March 1733 (SOY0325 Gregorian)`
 
## Dual Dating

"Dual dating" is a method of representing a date to include two start-of-year conventions simultaneously. It *does not* contain representations that touch on whether the Julian or Gregorian calendar is used (one or the other calendar is assumed, or otherwise indicated).

So using my previous example the Julian dates could be represented as `3rd March 1733/4`: 1733 (SOY0325) being written before 1734 (SOY0101).

I'm quoting Spathaky's [2011](http://www.cree.name/genuki/dates.htm) representation ...
> 3rd March 1733/4 

His further notes about the convention ...

> An oblique stroke is by far the most usual indicator, but sometimes the alternative final figures of the year are written above and below a horizontal line, as in a fraction (see [photograph](http://www.cree.name/genuki/gravestone.jpg)). Very occasionally a hyphen is used, as 1733-34.  [Spathaky 2011](http://www.cree.name/genuki/dates.htm)

Also ...
> ... whatever notation is used for the dual dating, it only has meaning in the period 1st January to 24th March each year (and of course for England and its colonies only before 1752). From 25th March to 31st December the Historical Year is the same as the Civil Year number so no special indication is necessary. [Spathaky 2011](http://www.cree.name/genuki/dates.htm)

In short and in general, dual dating is only useful (if it is useful at all) when referencing dates:
* Before the introduction of the Historical Year (SOY0101); and
* Between the Historical start-of-year (01-01) and the Civil start-of-year (whatever that is, e.g. 03-25).

## Prior standards confusion and modern conversions

Even at the time confusion reigned supreme. Not only did different countries adopt different standards at different times, but some countries settled those two issues at different times e.g. 

* England and its colonies adopted the switch from the Julian to Gregorian AND the start-of-year switch (to 1st Jan) in 1752; but, by contrast
* Scotland switched its start-of-year (to 1st Jan) in 1600, and switched from Julian to Gregorian in 1752 ([Spathaky 2011](http://www.cree.name/genuki/dates.htm)).

Compounding this is the confusion and/or ambiguity from individual original writers (understandable given the general confusion and ambiguity around contemporaneous conventions). Spathaky [2011](http://www.cree.name/genuki/dates.htm) notes ...
> Many [original] writers of course, especially in dating letters, gave no indication of whether they were using the Civil or the Historical Year and we are left to decide using contextual clues or, as Cheney wrote, "according to the nationality and the circumstances of the writer."

We are left with an interpretation and conversion task.  Spathaky [2011](http://www.cree.name/genuki/dates.htm) seems to express both the scholarly responsibility and general procedure well ....
> We as historians have no excuse for creating ambiguity ...  It is no good writing simply 20th January 1745, for a reader is left wondering whether we have used the Civil or the Historical Year. ... In transcribing original archives we should of course transcribe all dates exactly as written. ...  Interpretation should always be clearly separate from transcription.

> [Again and out of order] ... we are left to decide using contextual clues or, as Cheney wrote, "according to the nationality and the circumstances of the writer." [Spathaky 2011](http://www.cree.name/genuki/dates.htm)

Without disambiguation in the source date our chief contextual clue will be to match the nationality of the writer, or the nationality of where the writing took place, and the approximate date against the following kind of table:

Table: Gregorian/Julian and start-of-year Calendar changes, for each region.

| Region                                         | Last Civil Year Introduction (MMDD) | Last Historical Year Introduction (0101) | Gregorian Introduced                                                                       | Source                                                 |
|------------------------------------------------|------------------------------------------|-------------------------------------|--------------------------------------------------------------------------------------------|--------------------------------------------------------|
| Britain and colonies (including North America) | 12th Century/SOY0325                     | 1st Jan 1752 (Julian)               | "2nd September 1752 [Julian] was followed by 14th September [1752 Gregorian]".             | [Spathaky 2011](http://www.cree.name/genuki/dates.htm), https://en.wikipedia.org/wiki/Old_Style_and_New_Style_dates |
| Scotland                                       | ?                                        | 1600                                | "it was not until 1752, along with England, that Scotland adopted the Gregorian calendar". | [Spathaky 2011](http://www.cree.name/genuki/dates.htm) |

... it would be therefore up to us, as historical scholars, to:

* Transcribe (and preserve) the original date as found in the source document;
* Interpret the original date with respect to both: Gregorian V Julian calendars; and the start-of-year.
* Convert that, in the biblatex context, to an EDTF date.

I'll mention further recommendations about this in a biblatex context below.

## Old Style and New Style  

"Old Style and New Style" is a convention attempting to disambiguate dates. But you'd never use it yourself as ...

> the phrases Old Style and New Style have not been used in a consistent way. In eighteenth century usage in the English-speaking world, they referred to the combination of the two changes which occurred in England in 1752, namely the change to the start of the year and the change from the Julian to the Gregorian calendar. Later, some historians would convert pre-1752 Julian dates to their Gregorian equivalents and might use the phrase New Style to indicate that they had done this.

> We thus have two clearly differentiated concepts, the change of the date of New Year and the change from the Julian to the Gregorian calendar, but no terms which have been consistently used to refer to them. [Spathaky 2011](http://www.cree.name/genuki/dates.htm)

## Biblatex recommendations

(Gregorian V Julian) and start-of-year issues are distinct. But because they are often conflated we'd do well to consider them at the same time. So the first suggestion is to Nick to change the thread title to something like "Convert EDTF/Gregorian calendar dates to Julian calendar dates? Start of year issues."

I recommend, in the bibalatex core:

* Implement Nick's suggestion to 'Optionally add a label such as “(Julian)”' as on option (in addition to the metainformatin test `\if<datetype>julian` and `\ifdatejulian` that already exists.)

I recommend that users be encouraged, bibalatex.pdf, to:

* Never use Old Style and New Style. The convention has been used inconsistently and it is ambiguous.

* Never use dual dating representations. Civil start-of-year (e.g. from 25th Mar) conventions are entirely confusing. It is better to be aware that source dates may be represented with a Civil start-of-year (or even represented using Dual Dating), convert to EDTF, and output as a single year (which might be as a Julian) that starts from the Historical start-of-year (1st Jan), e.g. `3rd March 1733 (Julian)`.  The source date, which might be represented as as a Civil start-of-year (e.g. from 03-25) and Julian, is best transcribed to `note` field and disambiguated (more about this below).

* Know that the onus is on them to interpret the source date. That is, determine:
    - Whether it is a Julian date and, 
    - If it expressed with a Civil start of year, that is, if the approximate date is
        * Before the introduction of the Historical Year (SOY0101); and
        * Between the Historical start-of-year (01-01) and the Civil start-of-year (whatever that is, e.g. 03-25).

* For their own purposes (as when creating a string in the `note` field) use a start-of-year disambiguator, in SOYMMDD format. E.g `SOY0101`, `SOY0325`

* Behold the four example representations of one date, for their own purposes (as when creating a string in the `note` field). E.g.

    * ` 3rd March 1734 (SOY0101 Julian)` 
    * ` 3rd March 1733 (SOY0325 Julian)` (Without the disambiguiators this would probably be the representation in the source document. The Civil start of year has not yet kicked on, so we are still in '1733' in Civil year terms, but we are in '1743' in Historical year terms.)
    * `14th March 1734 (SOY0101 Gregorian)`
    * `14th March 1733 (SOY0325 Gregorian)`
 

* Use the `note` field to store the string representing the date as it appears in the source document, with Julian/Gregorian and start-of-year disambiguators. E.g.:

        @book{my_key
            ...
            note= {Date as it appears in document "3rd March 1734" (SOY0325 Julian)},
            date = {1733-03-17},  % Converted from manually by user into EDTF (SOY0101 Gregorian)
            ...  

* Point to an online Julian to Gregorian Calendar converter (as an example resource).

* Promote the use of "Table: Gregorian/Julian and start-of-year Calendar changes, for each region" (as above). (I probably ought tidy up https://en.wikipedia.org/wiki/Talk:Old_Style_and_New_Style_dates and put that table there: but if anyone wants to beat me to it, go ahead).


For example, as a biblatex user:

* I come across a document that has a date `3rd March 1733`.
* I observe the document was written by an English author. So I look up "Table: Gregorian/Julian and start-of-year Calendar changes, for each region" (on some wikipedia page) for England.
* I judge it likely that `3rd March 1733` is implicitly `3rd March 1733 (SOY0325 Julian)`. So I set `note= {Date as it appears in document "3rd March 1733" (SOY0325 Julian)}`
* I mentally convert this to `3rd March 1734 (SOY0101 Julian)`.
* I use an online Julian to Gregorian resource to convert `3rd March 1734 (SOY0101 Julian)` into `14th March 1734 (SOY0101 Gregorian)`. So I set `date = {1734-03-14}`
* Based on partly arbitrary personal preferences I set my biblatex options to output `julian=true` and `gregorianstart=1752-09-14` (for England).  My date, therefore, outputs as `1734-03-03 (Julian)` (or, when `alldates=short`... `03/03/1734 (Julian)`).

Example summary:

* Source date: `3rd March 1733`. Set `note= {Date as it appears in document "3rd March 1733" (SOY0325 Julian)}`
* EDFT date: Set `date = {1734-03-14}` % Implicitly (SOY0101 Gregorian).
* Output:  `1734-03-03 (Julian)` % Implicitly (SOY0101).

**Comment by plk** [comment]
I am inclinded to think that perhaps this, valuable though it undoubtedly is for those working in specific areas, is not fundamentally suited to inclusion in a general manual, overburdened as it already famously is with detail. The aim of the core software is to provide the tools but not to facilitate further than that. I would suppose that very few people at the moment would need even a fraction of the new functionality ...

**Comment by JohnLukeBentley** [comment]
You are probably right that we ought not include all that information in Biblatex.pdf. Moreover it's probably better if I (or someone else) code up a webpage that does the conversions (including Julian to Gregorian And SOY0325 to SOY0101 etc) and include the info there.

I would emphasize, however, that as infrequent as these cases may be, when they do arise they'll be unavoidable. That is, if you are wanting to store the date of publication of a source document that lists its date as `3rd March 1733` you necessarily face the interpretation task. You'll need to confront the questions "Is that a Julian date?"; and "Is that a SOY0325 date"?

Perhaps, as a temporary place holder, you might like to link directly to my post (or the whole thread) from biblatex.pdf. To aid that I've provided a title for the comment "The relationship between Gregorian to Julian conversions; Start of year conversions; Old Style and New Style; and Dual Dating". 

Perhaps also biblatex.pdf could bear a summary of these issues. For example something like ...

> Given all dates in Biblatex are stored in EDTF you may need to convert a date as it appears in a source document. That could involve both a conversion from Julian to Gregorian and a conversion from Civil Year (a year figured to start at some date other than 1st Jan) to Historical Year (a year figured to start on the 1st Jan). For example in Britain and its colonies before 1752 both conversions may be needed.

> If you do make a conversion best practice is to, in the `note` field, quote the date as it appears in the source document and mention any interpretations you make. Best practice is not use "Old Style and New Style" conventions nor "Dual Dating" conventions. 

> For example:
> * Source date: `3rd March 1733`. Set note= {Date as it appears in document "3rd March 1733" (SOY0325 Julian)} % SOY0325 = Start of year on March the 25th.
*  EDFT date: Set `date = {1734-03-14}` % Implicitly (SOY0101 Gregorian).
* Output: `1734-03-03 (Julian)` % Implicitly (SOY0101).

> For further information see (The relationship between Gregorian to Julian conversions; Start of year conversions; Old Style and New Style; and Dual Dating)[https://github.com/plk/biblatex/issues/448#issuecomment-233291859]

Are you inclined to reject the one software suggestion I made?

> Implement Nick's suggestion to 'Optionally add a label such as “(Julian)”' as on option (in addition to the metainformatin test \if<datetype>julian and \ifdatejulian that already exists.)

**Comment by JohnLukeBentley** [comment]
See also [my latest comment in "Convert EDTF/Gregorian calendar dates to Julian calendar dates? #448"](https://github.com/plk/biblatex/issues/448#issuecomment-233662688).

## Design

* Seasons. Output left-to-right order. I'm not fussed about changing the status quo: Season before date.

  So Nick helpfully points to some style manual that have opposing recommendations...

  APA Manual, 6th Ed., Sect 6.28 does indeed (as Nick quotes) recommend ...

  > If the date is given as a season, give the year and the season, separated by a comma and enclosed in parentheses.

  Chicago Manual of Style, Sect 14.180, gives an example like ...
  > Boyan Jovanovic and Peter L. Rousseau, “Specific Capital and Technological Variety,” Journal of Human Capital 2 (Summer 2008): 135, doi:10.1086/590066. 

  But, interestingly, the APA's own example contradicts this (APA Manual, 6th Ed., "7.02 Books,Reference Books, and Book Chapters", Example 29, p205.) ...

  > Graham, G. (2005). Behaviorism. In E. N. Zalta (Ed.), The Stanford encyclopedia of philosophy (Fall 2007 ed.). Retrieved from http://plato.stanford.edu/entries/behaviorism/

  So, if we were just going of those two style guides, that might tip the issue in favour of seasons before dates. 
  
  At the moment I happen to personally favour the Chicago style guide in general: so I don't want to make it harder for myself by arguing in favour of date-before-season (even though date-before-season would be my preference if I was constructing a custom style guide).


## 96-dates.tex suggestions.

* Your "times" dates already test for timezones. But I prefer to not conflate that test with date ranges. That is, I have (in addition)

        @misc{jlbdate130,
          note      = {Timezones},
          date      = {2004-04-25},                 % Date only.      
          origdate  = {2004-04-25T14:34:00},        % Date with time, floating timezone
          eventdate = {2004-04-25T14:34:00Z},       % Date with time, UTC timezone
          urldate   = {2004-04-25T14:34:00+05:00}   % Date with time, local timezone
        }

   Add or ignore, as you wish.

## Biblatex.pdf Errors

 * `time` option. 
 
    * page 53. "14:03:23UTC" to "14:03:23Z".
    * page 54. "Timezones, if present, are either 'UTC' ..." to "Timezones, if present, are either 'Z' ...

## Bugs

* Times are output in italic (while the Date part is in normal font). Expect Date and time to be output in normal font.

* Seasons. Abbreviated seasons are output when `alldates=ymd` and `abbreviate=false`.

        % Get
        date = Spr. 1723

        % Expect
        date = Spring 1723

    My english.lbx file looks good (so the problem doesn't seem to lie there) ...
 
        ...
        spring           = {{Spring}{Spr\adddot}},
        summer           = {{Summer}{Sum\adddot}},
        autumn           = {{Autumn}{Aut\adddot}},
        winter           = {{Winter}{Win\adddot}},

    Again, I'll probably personally override the season localisation string to always display the long version ("Summer" rather than "Sum."). Especially given that the Style manuals use the long version. But, bug aside, the status quo makes sense:  provide abbreviated V long versions and have this determined by the `abbreviate` option.


## Sytle: Authoryear Bug

I know I'm jumping the gun a bit, in that any packaged style issues properly come after we've stablised the core-testing 96-dates.tex, but there is the following bug you'll be keen to look at ...

* When author has multiple works in one year, bibliographic entries wrongly display double alphabetic enumerator. E.g. 'aa' instead of 'b'.
        
        % biblatex options
        \usepackage[style=authoryear,%
        % julian=true,% convert dates before gregorianstart to Julian Calendar
        % gregorianstart=1582-10-15,% Default
        %alldates=short,%
        alldates=ymd,% Year-Month-Day format
        %alldates=edtf,% prints EDTF format
        alldatesusetime=true, % print time components in non-compact date ranges
        %alltimes=12h,%
        alltimes=24h,% prints 24 hour format times (default)
        % alltimes=24hcomp,% prints 24 hour compressed format times
        % seconds=true,% uncomment to print seconds
        timezones=true,% uncomment to print timezones
        datezeros=true,%
        dateera=secular,% prints BCE/CE controlled by dateerauto, datezeros
        % dateera=christian,% prints BC/AD controlled by dateerauto, datezeros
        % dateera=astronomical,% prints astronomical dates, controlled by datezeros
        dateeraauto=600,% Sets the max year ceiling for automatic printing of era
        % abbreviate=false,% prints full date localisation strings
        dateuncertain=true,%
        datecirca=true,%
        sorting=none,%
        backend=biber]{biblatex}

        % Bib entries
        @article{russell_1914_nature,
        author = {Russell, Bertrand},
        date = {1914},
        title = {On the {{Nature}} of {{Acquaintance}}},
        pages = {1--16,161,435--453},
        journaltitle = {Monist},
        volume = {24}
        }
        @article{russell_1914_relation,
        author = {Russell, Bertrand},
        date = {1914},
        title = {The {{Relation}} of {{Sense-Data}} to {{Physics}}},
        pages = {1--27},
        journaltitle = {Scientia},
        volume = {16}
        }

        % Output
        Russell, Bertrand (1914aa). “On the Nature [...]
        — (1914bb). “The Relation of Sense-Data [...]

        % Expect
        Russell, Bertrand (1914a). “On the Nature [...]
        — (1914b). “The Relation of Sense-Data [...]

**Comment by plk** [comment]
The emphasis on the times is not part of biblatex, it's just done that way in that test file to make them stand out. Season abbreviations are controlled by `dateabbrev` and not `abbreviation`. The double `extrayear` was a bug and should be fixed in the latest dev upload.

**Comment by JohnLukeBentley** [comment]
plk wrote a comment in [Circa dates, circa date ranges, and question marked dates. Plus Eras. #427](https://github.com/plk/biblatex/issues/427#issuecomment-234074200):
> I really don't want to add such domain specific information to the biblatex manual - they are all issues which only very particular scholars will care about and their caring will likely manifest in knowing something about all that already one would think ...

There's probably a good treatment of these issues somewhere but on the cursory research I've done there seems to be a great deal of confusion and conflation about. So it could well be something the scholars in the relevant areas should do well, but fail to. I hope my post, on the back of Nick's, lends some clarity here.

But even on that assumption (that my post is helpful on those issues) my post is draft; and needs it's own webpage somewhere. And it is, partly by design, in some sense independent of biblatex. Moreover, I generally endorse gatekeepers, such as your self, doing their best to cut down on what is not strictly necessary. So if I ever get around to creating that webpage I might attempt to persuade you to point to it. Otherwise, I defer to your decision.

plk wrote a comment in [Circa dates, circa date ranges, and question marked dates. Plus Eras. #427](https://github.com/plk/biblatex/issues/427#issuecomment-234052235):
> If you mean the "(Julian)" label - I think that's a style issue not to be addressed in core. However, the testing - yes please, the more the better.

Given that you've given us the means, as users or style authors, to include the "(Julian)" label I'm not fussed that you should decide against putting it in the core.

So, from me and for the moment, I consider issues around *The relationship between Gregorian to Julian conversions; Start of year conversions; Old Style and New Style; and Dual Dating* closed.  

Nick, any comment?
